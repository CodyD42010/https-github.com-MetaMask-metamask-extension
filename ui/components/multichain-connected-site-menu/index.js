export { MultichainConnectedSiteMenu } from './multichain-connected-site-menu.component';
