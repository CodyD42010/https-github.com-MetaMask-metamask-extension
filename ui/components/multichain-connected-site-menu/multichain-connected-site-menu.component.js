import React from 'react';
import PropTypes from 'prop-types';
import {
  BorderColor,
  BorderRadius,
  Color,
  IconColor,
  Size,
} from '../../helpers/constants/design-system';
import { BadgeWrapper, Icon, ICON_NAMES } from '../component-library';
import Box from '../ui/box';

export const MultichainConnectedSiteMenu = ({ connectedSubjects }) => {
  return (
    <Box>
      {connectedSubjects.length ? (
        <BadgeWrapper
          badge={
            <Box
              backgroundColor={Color.successDefault}
              borderRadius={BorderRadius.full}
              borderColor={BorderColor.borderMuted}
              borderWidth={2}
              style={{ width: 12, height: 12 }}
            />
          }
        >
          <Icon
            name={ICON_NAMES.GLOBAL}
            size={Size.XL}
            color={IconColor.iconAlternative}
          />
        </BadgeWrapper>
      ) : (
        <BadgeWrapper
          badge={
            <Box
              backgroundColor={Color.backgroundDefault}
              borderRadius={BorderRadius.full}
              borderColor={BorderColor.borderMuted}
              borderWidth={2}
            />
          }
        >
          <Icon
            name={ICON_NAMES.GLOBAL}
            size={Size.XL}
            color={IconColor.iconAlternative}
          />
        </BadgeWrapper>
      )}
    </Box>
  );
};

MultichainConnectedSiteMenu.propTypes = {
  connectedSubjects: PropTypes.arrayOf(PropTypes.object).isRequired,
};
