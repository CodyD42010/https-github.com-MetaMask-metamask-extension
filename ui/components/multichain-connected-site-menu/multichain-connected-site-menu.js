import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { useSelector } from 'react-redux';
import {
  BorderColor,
  BorderRadius,
  Color,
  IconColor,
  Size,
} from '../../helpers/constants/design-system';
import { BadgeWrapper, Icon, ICON_NAMES } from '../component-library';
import Box from '../ui/box';
import { getConnectedSubjectsForSelectedAddress } from '../../selectors';

export const MultichainConnectedSiteMenu = ({ className }) => {
  const connectedSubjects = useSelector(getConnectedSubjectsForSelectedAddress);

  return (
    <Box
      className={classNames('multichain-connected-site-menu', className)}
      data-testid="connection-menu"
    >
      {connectedSubjects?.length ? (
        <BadgeWrapper
          badge={
            <Box
              backgroundColor={Color.successDefault}
              borderRadius={BorderRadius.full}
              borderColor={BorderColor.borderMuted}
              borderWidth={4}
              style={{ width: 14, height: 14 }}
            />
          }
        >
          <Icon
            name={ICON_NAMES.GLOBAL}
            size={Size.XL}
            color={IconColor.iconAlternative}
          />
        </BadgeWrapper>
      ) : (
        <BadgeWrapper
          badge={
            <Box
              backgroundColor={Color.backgroundDefault}
              borderRadius={BorderRadius.full}
              borderColor={BorderColor.borderMuted}
              borderWidth={4}
              style={{ width: 14, height: 14 }}
            />
          }
        >
          <Icon
            name={ICON_NAMES.GLOBAL}
            size={Size.XL}
            color={IconColor.iconAlternative}
          />
        </BadgeWrapper>
      )}
    </Box>
  );
};

MultichainConnectedSiteMenu.propTypes = {
  className: PropTypes.string,
};
