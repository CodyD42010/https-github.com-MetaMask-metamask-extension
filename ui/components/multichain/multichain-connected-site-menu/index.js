export { MultichainConnectedSiteMenu } from './multichain-connected-site-menu';
