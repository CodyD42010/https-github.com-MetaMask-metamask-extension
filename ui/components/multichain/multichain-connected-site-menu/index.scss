.multichain-connected-site-menu {
  &__badge {
    height: 16px;
    width: 16px;
  }

  &__badge.not-connected {
    height: 10px;
    width: 10px;
  }

  &__badge.not-connected::after {
    content: '';
    position: absolute;
    top: -3px;
    left: -3px;
    right: -3px;
    bottom: -3px;
    background: var(--color-background-default);
    z-index: -1;
    border-radius: 50%;
  }
}
