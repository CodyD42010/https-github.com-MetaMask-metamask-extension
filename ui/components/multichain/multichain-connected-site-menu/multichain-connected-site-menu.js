import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { useSelector } from 'react-redux';
import {
  STATUS_CONNECTED_TO_ANOTHER_ACCOUNT,
  STATUS_NOT_CONNECTED,
} from '../../../helpers/constants/connected-sites';
import {
  BackgroundColor,
  BorderColor,
  BorderRadius,
  IconColor,
  Size,
} from '../../../helpers/constants/design-system';
import { BadgeWrapper } from '../../component-library';
import { Icon, ICON_NAMES } from '../../component-library/icon/deprecated';
import Box from '../../ui/box';
import { getSelectedIdentity } from '../../../selectors';
import Tooltip from '../../ui/tooltip';
import { useI18nContext } from '../../../hooks/useI18nContext';

export const MultichainConnectedSiteMenu = ({
  className,
  globalMenuColor,
  status,
  text,
}) => {
  const t = useI18nContext();
  const selectedAccount = useSelector(getSelectedIdentity);
  return (
    <Box
      className={classNames('multichain-connected-site-menu', className)}
      data-testid="connection-menu"
    >
      <Tooltip
        title={
          status === STATUS_NOT_CONNECTED
            ? t('statusNotConnectedAccount')
            : `${selectedAccount?.name} ${text}`
        }
        data-testid="multichain-connected-site-menu__tooltip"
        position="bottom"
      >
        <BadgeWrapper
          positionObj={
            status === STATUS_CONNECTED_TO_ANOTHER_ACCOUNT
              ? { bottom: 4, right: -1, zIndex: 1 }
              : { bottom: 2, right: -4, zIndex: 1 }
          }
          badge={
            <Box
              backgroundColor={globalMenuColor}
              className={`multichain-connected-site-menu__badge ${
                status === STATUS_CONNECTED_TO_ANOTHER_ACCOUNT
                  ? 'not-connected'
                  : ''
              }`}
              borderRadius={BorderRadius.full}
              borderColor={
                status === STATUS_CONNECTED_TO_ANOTHER_ACCOUNT
                  ? BorderColor.successDefault
                  : BackgroundColor.backgroundDefault
              }
              borderWidth={
                status === STATUS_CONNECTED_TO_ANOTHER_ACCOUNT ? 2 : 3
              }
            />
          }
        >
          <Icon
            name={ICON_NAMES.GLOBAL}
            size={Size.LG}
            color={IconColor.iconDefault}
          />
        </BadgeWrapper>
      </Tooltip>
    </Box>
  );
};

MultichainConnectedSiteMenu.propTypes = {
  /**
   * Additional classNames to be added to the MultichainConnectedSiteMenu
   */
  className: PropTypes.string,
  /**
   * Background color based on the connection status
   */
  globalMenuColor: PropTypes.string.isRequired,
  /**
   * Connection status string
   */
  status: PropTypes.string.isRequired,
  /**
   * Connection status message
   */
  text: PropTypes.string,
};
