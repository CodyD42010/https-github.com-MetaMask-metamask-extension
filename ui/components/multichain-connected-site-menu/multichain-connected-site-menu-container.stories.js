import React from 'react';
import testData from '../../../.storybook/test-data';
import { MultichainConnectedSiteMenu } from './multichain-connected-site-menu.component';

const connectedSubjectsMetaData = Object.values(
  testData.metamask.subjectMetadata,
);

export default {
  title: 'Components/UI/MultichainConnectedSiteMenu',

  component: MultichainConnectedSiteMenu,
  argTypes: {
    connectedSubjects: {
      control: 'array',
    },
  },
  args: {
    connectedSubjects: [],
  },
};

export const DefaultStory = (args) => <MultichainConnectedSiteMenu {...args} />;

DefaultStory.storyName = 'Default';

export const ConnectedStory = (args) => (
  <MultichainConnectedSiteMenu {...args} />
);

ConnectedStory.args = {
  connectedSubjects: connectedSubjectsMetaData,
};
