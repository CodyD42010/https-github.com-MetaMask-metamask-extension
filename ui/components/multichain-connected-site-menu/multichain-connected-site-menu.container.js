import { connect } from 'react-redux';

import { getConnectedSubjectsForSelectedAddress } from '../../selectors';

import { MultichainConnectedSiteMenu } from './multichain-connected-site-menu.component';

const mapStateToProps = (state) => {
  const connectedSubjects = getConnectedSubjectsForSelectedAddress(state);

  return {
    connectedSubjects,
  };
};

export default connect(mapStateToProps)(MultichainConnectedSiteMenu);
