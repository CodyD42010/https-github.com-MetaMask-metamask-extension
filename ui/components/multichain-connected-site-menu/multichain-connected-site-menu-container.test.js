import React from 'react';
import configureMockStore from 'redux-mock-store';
import { renderWithProvider } from '../../../test/jest';
import { MultichainConnectedSiteMenu } from './multichain-connected-site-menu.component';

describe('Multichain Connected Site Menu', () => {
  const mockStore = {
    metamask: {
      selectedAddress: '0x8e5d75d60224ea0c33d0041e75de68b1c3cb6dd5',
    },
  };
  describe('render', () => {
    it('should match snapshot for not connected state', () => {
      const store = configureMockStore()(mockStore);
      const { getByTestId, container } = renderWithProvider(
        <MultichainConnectedSiteMenu />,
        store,
      );
      expect(getByTestId('connection-menu')).toBeDefined();
      expect(container).toMatchSnapshot();
    });

    it('should match snapshot for connected state', () => {
      const connectedState = {
        metamask: {
          ...mockStore.metamask,
          subjectMetadata: {
            'peepeth.com': {
              iconUrl: 'https://peepeth.com/favicon-32x32.png',
              name: 'Peepeth',
            },
          },
          subjects: {
            'peepeth.com': {
              permissions: {
                eth_accounts: {
                  caveats: [
                    {
                      type: 'restrictReturnedAccounts',
                      value: ['0x8e5d75d60224ea0c33d0041e75de68b1c3cb6dd5'],
                    },
                  ],
                  date: 1585676177970,
                  id: '840d72a0-925f-449f-830a-1aa1dd5ce151',
                  invoker: 'peepeth.com',
                  parentCapability: 'eth_accounts',
                },
              },
            },
          },
        },
      };
      const store = configureMockStore()(connectedState);
      const { getByTestId, container } = renderWithProvider(
        <MultichainConnectedSiteMenu />,
        store,
      );
      expect(getByTestId('connection-menu')).toBeDefined();
      expect(container).toMatchSnapshot();
    });
  });
});
