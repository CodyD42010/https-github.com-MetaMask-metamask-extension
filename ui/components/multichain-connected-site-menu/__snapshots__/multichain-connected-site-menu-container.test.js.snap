// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Multichain Connected Site Menu render should match snapshot 1`] = `
<div>
  <div
    class="box multichain-connected-site-menu box--flex-direction-row"
    data-testid="connection-menu"
  >
    <div
      class="box mm-badge-wrapper box--display-inline-block box--flex-direction-row"
    >
      <span
        class="box mm-icon mm-icon--size-xl box--display-inline-block box--flex-direction-row box--color-icon-alternative"
        style="mask-image: url('./images/icons/global.svg');"
      />
      <div
        class="box mm-badge-wrapper__badge-container mm-badge-wrapper__badge-container--circular-top-right box--flex-direction-row"
      >
        <div
          class="box box--flex-direction-row box--background-color-background-default box--rounded-full box--border-color-border-muted box--border-width-2 box--border-style-solid"
        />
      </div>
    </div>
  </div>
</div>
`;
